confluent/docker_utils/cub.py

```diff
@@ -229,6 +230,37 @@ def check_kafka_rest_ready(host, port, service_timeout):
         return False
 
 
+def check_connect_ready(host, port, service_timeout):
+    """Waits for Connect to be ready.
+
+    Args:
+        host: Hostname where Connect worker is hosted.
+        port: Connect port.
+        timeout: Time in secs to wait for the service to be available.
+
+    Returns:
+        False, if the timeout expires and Connect is not ready, True otherwise.
+
+    """
+
+    # Check if you can connect to the endpoint
+    status = wait_for_service(host, port, service_timeout)
+
+    if status:
+        # Check if service is responding as expected to basic request
+        url = "http://%s:%s" % (host, port)
```

Comment:
i see we use just http elsewhere, but how does this work for secured connect clusters (i.e. SSL enabled on the REST API)?

Reply:
@ewencp good point, and it also applies to current implementation of `sr-ready` if SR is enabled only for https.

Question is whether this PR should match current assumptions and behavior of other `*-ready` commands, or should it be better and more robust?

Except for `cp-demo`, I don't use https in any other demos, so I'm ok with providing matching current assumptions and behavior of`*-ready` commands.

Reply:
tbh, not sure off the top of my head. I wasn't involved in review of any of the existing code.

I agree that matching expectations of existing code is probably the right baseline. Part of why I called it out is b/c I am unclear how this works today for (e.g. SR) in a secured environment, which presumably we have users implementing.

perhaps a stretch, but @arrawatia since I think you originated this code way back when, are we missing something re: assumptions, or is this not intended to work with https, or something else? or alternatively, who could answer this?

Reply:
@ewencp I filed a separate jira to track https.  For now I propose we merge this PR as-is, with the understanding that it works only for http.